WooCommerce/Classes/ViewRelated/Blaze/AdDestination/BlazeAdDestinationSettingViewModel.swift

```diff
@@ -27,23 +32,48 @@ final class BlazeAdDestinationSettingViewModel: ObservableObject {
         return String(format: Localization.finalDestination, buildFinalDestinationURL())
     }
 
+    // View model for the add parameter view.
+    var blazeAddParameterViewModel: BlazeAddParameterViewModel {
+        let urlParameter = selectedParameterIndex != nil ? parameters[selectedParameterIndex!] : nil
+
+        return BlazeAddParameterViewModel(
+            remainingCharacters: calculateRemainingCharacters(),
+            isNotFirstParameter: parameters.count > 1,
+            parameter: urlParameter,
+            onCancel: { self.clearSelectedParameterIndex() },
```

Comment:
I wasn't sure about this part at first, and when I asked chatGPT, it says:

>  in your onCancel closure, you're not capturing self in a way that would cause a retain cycle. The clearSelectedParameterIndex() function call will be executed and then the closure will exit, releasing the strong reference to self.

I should've dug deeper and don't immediately trust it. 🤔 

Added in https://github.com/woocommerce/woocommerce-ios/pull/11642/commits/dab1394cd2869e231d54c60e828a8fa385a5e0dd

Reply:
I would not rely too much on ChatGPT. When you're in doubt, please ask us. We're always there to answer questions.